In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("Coffee.csv")

In [3]:
df.shape

(520, 8)

In [4]:
df

,Unnamed: 0,Year,Country,Domestic Consumption,Exports,Import Price,Export Price,Total Production
0,0,2015,Argentina,13051.0,14887.0,2.6,2.7,26614.0
1,1,2015,Argentina,13131.0,15055.0,2.6,3.0,27019.0
2,2,2015,Argentina,13305.0,15075.0,2.7,NaN,27232.0
3,3,2015,Brazil,21070.0,36806.0,1.9,1.9,54721.0
4,4,2015,Canada,18063.0,21734.0,3.9,4.0,38987.0
...,...,...,...,...,...,...,...,...
515,515,2023,USA,28629.0,0.0,5.3,0.0,0.0
516,516,2023,USA,28875.0,0.0,5.2,0.0,0.0
517,517,2023,Vietnam,NaN,34342.0,2.9,3.2,52222.0
518,518,2023,Vietnam,18557.0,34695.0,3.2,2.9,52867.0


1) How many rows and columns are present in the dataset?
ans: There are 520 rows and 8 columns

In [6]:
df.columns

Index(['Unnamed: 0', 'Year', 'Country', 'Domestic Consumption', 'Exports',
       'Import Price', 'Export Price', 'Total Production'],
      dtype='object')

2) What are the names of all columns in the dataset?
Ans: The columns name are 1.Unnamed: 0 2.Year 3.Country 4.Domestic Consumption 5.Exports 6.Import Price 7.Export Price 8.Total Production

3) What is the data type of each column?

In [7]:
df.dtypes

Unnamed: 0                int64
Year                      int64
Country                  object
Domestic Consumption    float64
Exports                 float64
Import Price            float64
Export Price            float64
Total Production        float64
dtype: object

4) Display the first 5 rows of the dataset.

In [8]:
df.head()

,Unnamed: 0,Year,Country,Domestic Consumption,Exports,Import Price,Export Price,Total Production
0,0,2015,Argentina,13051.0,14887.0,2.6,2.7,26614.0
1,1,2015,Argentina,13131.0,15055.0,2.6,3.0,27019.0
2,2,2015,Argentina,13305.0,15075.0,2.7,NaN,27232.0
3,3,2015,Brazil,21070.0,36806.0,1.9,1.9,54721.0
4,4,2015,Canada,18063.0,21734.0,3.9,4.0,38987.0


5) Display the last 5 rows of the dataset.

In [9]:
df.tail()

,Unnamed: 0,Year,Country,Domestic Consumption,Exports,Import Price,Export Price,Total Production
515,515,2023,USA,28629.0,0.0,5.3,0.0,0.0
516,516,2023,USA,28875.0,0.0,5.2,0.0,0.0
517,517,2023,Vietnam,NaN,34342.0,2.9,3.2,52222.0
518,518,2023,Vietnam,18557.0,34695.0,3.2,2.9,52867.0
519,519,2023,Vietnam,18615.0,34840.0,3.0,NaN,52639.0


6) Which columns contain missing values?

In [10]:
df.isnull().sum()

Unnamed: 0               0
Year                     0
Country                  0
Domestic Consumption    26
Exports                 26
Import Price            26
Export Price            26
Total Production        26
dtype: int64

5 columns contain 26 missing values each .the columns are 'Domestic Consumption','Exports','Import Price','Export Price','Total Production'

In [13]:
df["Country"].nunique()

20

In [12]:
df["Country"].unique()

array(['Argentina', 'Brazil', 'Canada', 'Chile', 'China', 'Colombia',
       'Ethiopia', 'France', 'Germany', 'India', 'Indonesia', 'Italy',
       'Kenya', 'Mexico', 'Peru', 'South Africa', 'Thailand', 'Turkey',
       'USA', 'Vietnam'], dtype=object)

7) How many unique countries are present in the dataset?
Ans: there are 20 unique countries .

8) How many records are available for each year?

In [14]:
df["Year"].value_counts().sort_index()

Year
2015    73
2016    58
2017    59
2018    44
2019    57
2020    58
2021    61
2022    56
2023    54
Name: count, dtype: int64

9) How many records are available for each country?

In [15]:
df["Country"].value_counts()

Country
France          31
Argentina       30
Colombia        30
South Africa    29
Chile           29
Germany         28
Mexico          28
Italy           28
Indonesia       27
India           26
China           26
USA             25
Canada          24
Turkey          24
Vietnam         24
Brazil          23
Ethiopia        23
Kenya           23
Thailand        22
Peru            20
Name: count, dtype: int64

10) Find the minimum, maximum, and average of total production.

In [17]:
print(df["Total Production"].min())
print(df["Total Production"].max())
print(df["Total Production"].mean())

0.0
264565.0
46466.72672064777


11) Which country has the highest total production in a single record?
Ans: China has the highest total production in single record

In [24]:
df.loc[df["Total Production"].idxmax()]

Unnamed: 0                   480
Year                        2023
Country                    China
Domestic Consumption    159839.0
Exports                 108888.0
Import Price                 3.9
Export Price                 3.8
Total Production        264565.0
Name: 480, dtype: object

In [25]:
df[(df["Total Production"].isnull()) | (df["Total Production"] == 0)]

,Unnamed: 0,Year,Country,Domestic Consumption,Exports,Import Price,Export Price,Total Production
14,14,2015,China,119776.0,85031.0,3.0,3.5,NaN
27,27,2015,Germany,8514.0,11030.0,3.2,3.5,0.0
28,28,2015,Germany,8514.0,11030.0,3.2,3.5,0.0
29,29,2015,Germany,8887.0,11273.0,3.4,3.6,0.0
30,30,2015,Germany,8396.0,10988.0,3.0,3.7,0.0
...,...,...,...,...,...,...,...,...
512,512,2023,USA,28601.0,0.0,5.5,0.0,0.0
513,513,2023,USA,28632.0,0.0,5.1,0.0,0.0
514,514,2023,USA,29342.0,0.0,5.5,0.0,0.0
515,515,2023,USA,28629.0,0.0,5.3,0.0,0.0


12) Are there any rows where total production is missing or zero?
Ans: There are 76 rows where total production is missing or valued 0

13) What is the average import price across all records?

In [26]:
df["Import Price"].mean()

3.1564777327935225

14) What is the total coffee production for each year?

In [27]:
df.groupby("Year")["Total Production"].sum()

Year
2015    2551154.0
2016    2163533.0
2017    2438153.0
2018    1793354.0
2019    2593378.0
2020    2756933.0
2021    2867085.0
2022    2681517.0
2023    3109456.0
Name: Total Production, dtype: float64

15) Which are the top 5 countries based on total production?

In [28]:
df.groupby("Country")["Total Production"].sum().sort_values(ascending=False).head(5)

Country
China        5682192.0
India        1902283.0
Mexico       1838210.0
France       1837554.0
Indonesia    1582361.0
Name: Total Production, dtype: float64

16) Which country has the highest average export price?

In [29]:
df.groupby("Country")["Export Price"].mean().sort_values(ascending=False).head(1)

Country
Canada    4.338095
Name: Export Price, dtype: float64

17) Does domestic consumption exceed total production for any country?

In [32]:
df[df["Domestic Consumption"] > df["Total Production"]]

,Unnamed: 0,Year,Country,Domestic Consumption,Exports,Import Price,Export Price,Total Production
27,27,2015,Germany,8514.0,11030.0,3.2,3.5,0.0
28,28,2015,Germany,8514.0,11030.0,3.2,3.5,0.0
29,29,2015,Germany,8887.0,11273.0,3.4,3.6,0.0
30,30,2015,Germany,8396.0,10988.0,3.0,3.7,0.0
32,32,2015,Germany,8121.0,11043.0,3.1,3.7,0.0
63,63,2015,USA,24902.0,NaN,4.6,0.0,0.0
64,64,2015,USA,25244.0,0.0,4.4,0.0,0.0
66,66,2015,USA,24831.0,0.0,4.2,0.0,0.0
94,94,2016,Germany,8792.0,11048.0,3.3,3.6,0.0
95,95,2016,Germany,8670.0,10868.0,3.2,3.7,0.0


Only Germany and USA are two countries where domestic consumption exceed total production

18) Which country shows the highest growth in total production over the years?

In [33]:
growth = df.sort_values("Year").groupby("Country")["Total Production"] .agg(lambda x: x.iloc[-1] - x.iloc[0])

growth.sort_values(ascending=False).head(1)

Country
China    64528.0
Name: Total Production, dtype: float64

China has the highest growth in total production over the years

19) Is there any relationship between export price and total production?

In [34]:
df["Export Price"].corr(df["Total Production"])

0.3442034661421095

The corelation point between Export Price and Total Production is 0.344 .It shows there is a positive relation between them but the relation is not that strong

20) Write 4–5 lines summarizing key insights from the EDA.


Ans:The dataset shows significant variation in coffee production across countries and years. China dominates total production and exhibits the highest growth trend. Some developed countries like Germany and USA rely heavily on imports, as their domestic production is zero. There is a moderate positive relationship between export price and production. Overall, production trends show growth over time, especially in leading producer countries.